policy/modules/contrib/virt.te

```diff
@@ -718,6 +718,20 @@ optional_policy(`
 	dnsmasq_manage_pid_files(virtd_t)
 ')
 
+optional_policy(`
+	passt_socket(virt_domain, user_tmp_t)
+	passt_socket(virt_domain, qemu_var_run_t)
+
+	passt_pidfile(virtd_t, user_tmp_t)
+	passt_pidfile(virtd_t, virt_var_run_t)
+	passt_pidfile(virtd_t, qemu_var_run_t)
+
+	passt_logfile(user_tmp_t)
```

Comment:
This is to let users configure passt to write a log file under `/var/run/<UID>/libvirt`, for current lack of a better type. It's a temporary file in some sense, it's expected to be used for debugging rather than for auditing.

Reply:
Not specifically related to the logfile, but I'm wondering... When running in `qemu:///system` mode, both the pidfile and the socket will be created under `/run/libvirt/qemu/passt/` (`qemu_var_run_t`). When running in `qemu:///session` mode, `virtqemud` will be running unconfined, so I would expect passt to also follow that. In other words, I think the `user_tmp_t` rules are unnecessary.

Now speaking of the logfile in particular :) The current XML schema allows the user to pick an arbitrary path where the file should be stored, but that's obviously problematic because, even with the rule you've added here, there are going to be a number of technically valid paths that are not going to work in practice. So I wonder, does it even make sense to make to allow the user to pick an arbitrary path as opposed to generating a predictable one?

Prior art for predictable locations: swtpm. Counter-example: serial console logging, where arbitrary paths are allowed.

Reply:
> When running in `qemu:///session` mode, `virtqemud` will be running unconfined

Not in my tests, and `/usr/sbin/virtqemud` is defined as `virtd_exec_t` by `virt.fc`.

> so I would expect passt to also follow that

Even if that was the case, we clearly don't want to let passt run unconfined.

> So I wonder, does it even make sense to make to allow the user to pick an arbitrary path as opposed to generating a predictable one?

Probably not, but here I'm reflecting the current status of the libvirt integration.

> Prior art for predictable locations: swtpm.

This looks preferable to me because it avoids tricking the user into set a path that can't actually be used.

Reply:
> > So I wonder, does it even make sense to make to allow the user to pick an arbitrary path as opposed to generating a predictable one?
> 
> Probably not, but here I'm reflecting the current status of the libvirt integration.

At the time I wrote the libvirt code, I was not aware that passt had selinux policies to limit where the logfile could be written :-/

> > Prior art for predictable locations: swtpm.
> 
> This looks preferable to me because it avoids tricking the user into set a path that can't actually be used.

Which aspect of swtpm are you talking about here? (I think we're straying a bit from discussing the job at hand in this PR, BTW...)

Reply:
>> When running in `qemu:///session` mode, `virtqemud` will be running unconfined
>>
> Not in my tests, and `/usr/sbin/virtqemud` is defined as `virtd_exec_t` by `virt.fc`.
>
>> so I would expect passt to also follow that
>>
> Even if that was the case, we clearly don't want to let passt run unconfined.

Mh, here `virtqemud` (in `qemu:///session` mode, as autostarted by running `virsh -c qemu:///session list` from a terminal) definitely runs unconfined. But QEMU itself runs as `svirt_t` and passt gets `passt_t`, just like in `qemu:///system` mode. I'm not sure how that's happening, but it looks like indeed we need those `user_tmp_t` rules.

With that established, let's leave the discussion about logfile to another day and forum :)